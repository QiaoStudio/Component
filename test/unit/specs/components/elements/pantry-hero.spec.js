import {
  shallowMount
} from 'dp-test'
import ComHero from 'components/elements/hero/hero.vue'
import sinon from 'sinon'
import _ from 'lodash'

describe('elements/Hero', () => {
  let wrapper
  let sanbox
  let imgSrc = 'https://picsum.photos/768/380'

  function createWrapper(options) {
    return shallowMount(ComHero, options)
  }

  function generateWrapper(options) {
    return createWrapper(_.merge({
      propsData: {
        isCurved: true,
        curveColor: '#FFFFFF',
        size: '',
        imageUrl: 'https://picsum.photos/1440/500?random',
        imageAlt: 'My Hero Image',
        overlayOpacity: ''
      }
    }, options || {}))
  }

  beforeEach(() => {
    sanbox = sinon.createSandbox()
    wrapper = generateWrapper()
  })

  afterEach(() => {
    wrapper.destroy()
    sanbox.restore()
  })

  it('when hero is initialized, hero should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('when hero size is set, hero background-image should be used', () => {
    wrapper.setProps({
      imageUrl: imgSrc,
      size: 'small'
    })
    expect(wrapper.find('.dp-hero').classes()).toContain('dp-hero--sm')
    wrapper.setProps({
      imageUrl: imgSrc,
      size: 'medium'
    })
    expect(wrapper.find('.dp-hero').classes()).toContain('dp-hero--md')
    wrapper.setProps({
      imageUrl: imgSrc,
      size: 'large'
    })
    expect(wrapper.find('.dp-hero').classes()).toContain('dp-hero--lg')
    expect(wrapper.find('.dp-hero__bg').exists()).toBe(true)
    expect(wrapper.find('.dp-hero__bg').attributes().style).toBe(`background-image: url(${imgSrc});`)
  })

  it('when hero curved is set, hero curve should be shown', () => {
    wrapper.setProps({
      isCurved: false,
      imageUrl: imgSrc
    })
    expect(wrapper.find('.dp-hero__curve').exists()).toBe(false)
    wrapper.setProps({
      isCurved: true,
      imageUrl: imgSrc
    })
    expect(wrapper.find('.dp-hero__curve').exists()).toBe(true)
  })

  it('when hero curved color is set, hero curve color should be correct', () => {
    let curveColor = '#000'
    wrapper.setProps({
      curveColor: curveColor,
      imageUrl: imgSrc
    })
    expect(wrapper.find('.dp-hero__curve path').attributes().fill).toBe(curveColor)
  })

  it('when hero opacity is set, hero opacity should be shown', () => {
    let opacity = '0.7'
    wrapper.setProps({
      overlayOpacity: opacity,
      imageUrl: imgSrc
    })
    expect(wrapper.find('.dp-hero__overlay').attributes().style).toBe(`opacity: ${opacity};`)
  })
})
