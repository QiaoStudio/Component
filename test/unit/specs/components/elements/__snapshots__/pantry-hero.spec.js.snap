// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/Hero when hero is initialized, hero should match snapshot 1`] = `
<div class="dp-hero dp-hero--auto">
  <div class="dp-hero__bg" style="background-image: url(https://picsum.photos/1440/500?random);">
    <div class="dp-hero__content"></div>
  </div>
  <!---->
  <div class="dp-hero__curve">
    <svg width="1440px" height="131px" viewBox="0 0 1440 131" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" preserveAspectRatio="xMidYMax">
      <path d="m0,0c458.78016,173.11964 980.65089,173.33387 1440,0l0,130.92008l-1440,0l0,-130.92008z" fill="#FFFFFF"></path>
    </svg>
  </div>
</div>
`;
